### ISSUE TITLE
GO_term usage 

### ISSUE THREAD

**Issue by Paul-Davis** [WormBase/website#5906]
There are a very small number of GO_term objects that are created/referenced each production cycle that are potentially invalid/need updating. Not a major issue but can be confusing in the web display as they are lacking most informative data.

Example: WS262

```
Ghosts created by one way XREF
---------------------------------
1) GO_term : "GO:0040023"
2) GO_term : "GO:0044767"
3) GO_term : "GO:0007067"

Working out where it's coming from example:
4) GO_term : "GO:0007126"
caltech_GO_defs.ace:GO_term : "GO:0007126" -O "2017-09-25_17:32:50_wormpub"
caltech_GO_defs.ace:Alt_id	 -O "2017-09-21_11:07:59_kimberly" "GO:0007126" -O "2017-09-21_11:07:59_kimberly"

caltech_Paper.ace:Refers_to	 -O "2006-05-05_12:49:27_andrei" GO_term -O "2017-09-21_13:44:00_wen" "GO:0007126" -O "citacePostProcess_inferred_from_WBProcess_wen"
caltech_Paper.ace:Refers_to	 -O "2017-09-21_11:06:12_chris" GO_term -O "2017-09-21_13:44:00_wen" "GO:0007126" -O "citacePostProcess_inferred_from_WBProcess_wen"
caltech_Paper.ace:Refers_to	 -O "2017-09-21_11:07:08_karen" GO_term -O "2017-09-21_13:44:00_wen" "GO:0007126" -O "citacePostProcess_inferred_from_WBProcess_wen"
caltech_Paper.ace:Refers_to	 -O "2017-09-21_11:07:08_karen" GO_term -O "2017-09-21_13:44:00_wen" "GO:0007126" -O "citacePostProcess_inferred_from_WBProcess_wen"
caltech_Paper.ace:Refers_to	 -O "2017-09-21_11:07:08_karen" GO_term -O "2017-09-21_13:44:00_wen" "GO:0007126" -O "citacePostProcess_inferred_from_WBProcess_wen"
caltech_Paper.ace:Refers_to	 -O "2017-09-21_11:07:08_karen" GO_term -O "2017-09-21_13:44:00_wen" "GO:0007126" -O "citacePostProcess_inferred_from_WBProcess_wen"

caltech_Phenotype.ace:Description	 -O "2017-09-21_11:07:04_gary" "Variations in the specialized nuclear and cytoplasmic divisions of a single diploid cell, whose specific outcome is the formation of four haploid daughter cells compared to control. Meiosis occurs during the formation of gametes." -O "2017-09-21_11:07:04_gary" GO_term_evidence -O "2017-09-21_11:07:04_gary" "GO:0007126" -O "2017-09-21_11:07:04_gary"

caltech_WBProcess.ace:Associated_with	 -O "2017-09-21_11:07:08_karen" GO_term -O "2017-09-21_11:07:08_karen" "GO:0007126" -O "2017-09-21_11:07:08_karen"

2-way XREF
-------------
GO_term : "GO:0005932"
Expr_pattern     "Expr2968"
Picture  "WBPicture0000009552"

GO_term : "GO:0016023"
Expr_pattern     "Expr549"
Picture  "WBPicture0000007955"

GO_term : "GO:0031513"
Expr_pattern     "Expr285"
Picture  "WBPicture0000012583"

GO_term : "GO:0035085"
Expr_pattern     "Expr2373"
Picture  "WBPicture0000011792"

GO_term : "GO:0051297"
GO_annotation    "00089814"
```

**Comment by draciti** [comment]
thanks Paul, i'll take a look at the ones coming from Expr_pattern

**Comment by draciti** [comment]
Added the GO updated terms as per below. Asked Juancarlos to delete the invalid terms in postgres (not visible in OA probably because invalid):

GO:0005932 replaced by GO:0036064; updated OA with the new term in Expr2968 and WBPicture0000009552.
Ref to GO replacement: http://amigo.geneontology.org/amigo/term/GO:0005932

GO:0016023 replaced by GO:0031410; updated OA with the new term in Expr549 and WBPicture0000007955
Ref to GO replacement: http://amigo.geneontology.org/amigo/term/GO:0016023

GO:0031513 replaced by GO:0097730; updated OA with the new term in Expr285 and WBPicture0000012583
Ref to GO replacement: http://amigo.geneontology.org/amigo/term/GO:0031513

GO:0035085 replaced by GO:0005930; updated OA with the new term in Expr2373 and WBPicture0000011792
Ref to GO replacement: http://amigo.geneontology.org/amigo/term/GO:0035085

**Comment by azurebrd** [comment]
I don't remember how your data works.  Why is it entered both through the picture OA and also through the expression OA instead of just one or the other ?  You can query for the pgids like : 
 SELECT * FROM pic_goid WHERE pic_goid ~ '0005932';
 joinkey |         pic_goid          |         pic_timestamp         
---------+---------------------------+-------------------------------
 8815    | "GO:0005932","GO:0035085" | 2011-08-02 10:27:33.018477-07

       exp_qualifiertext_hst     exp_species               
 SELECT * FROM exp_goid WHERE exp_goid ~ '0005932';
 joinkey |         exp_goid          |         exp_timestamp         
---------+---------------------------+-------------------------------
 6864    | "GO:0005932","GO:0035085" | 2011-05-31 12:11:47.633561-07

Then you can query the respective pgid through each OA, and edit the field with the correct data, since presumably, if I delete the whole entry, you'll lose the record of anything else that should still be associated.

**Comment by draciti** [comment]
1) data is curated in both picture and expression because the terms curated in picture can be a subset of what is in expression

2) As I previously wrote you in the e-mail, I need you to delete the obsolete GO terms (and not the whole entry) as they are not visible in the OA interface so I cannot edit them, but they are dumped in the .ace file

let me know how i can help

**Comment by vanaukenk** [comment]
@chris-grove @kyook @Gschindelman 

It looks like some of these obsolete GO terms are associated with WBProcess and Phenotype objects.  Can you check the data and make any necessary updates for the next upload?  

1) GO_term : "GO:0040023"
2) GO_term : "GO:0044767"
3) GO_term : "GO:0007067"
4) GO_term : "GO:0007126"

Let me know if you have any questions.  Thx.

**Comment by chris-grove** [comment]
@vanaukenk 
Yes, I'll take a look

**Comment by chris-grove** [comment]
@vanaukenk 
Using Postgres queries:

- I've found two Phenotype OA entries (PGIDs 51149 and 51150) that each have the app_goprocess table refer to GO:0044767. I'll work with @azurebrd to delete those from the app_goprocess table.

- I've found two Process Term OA entries (PGIDs 60 and 99) that refer, respectively, to GO:0007067 and to GO:0007126; I'll talk to @kyook and @azurebrd to remove those.

- The GO OA table gop_goid has ~60-80 entries with GO:0007067 and ~100-120 entries with GO:0007126; I'll let you take care of those

I could not find any phenotype or process annotations/entries referring to GO:0040023

**Comment by chris-grove** [comment]
@Paul-Davis 
@vanaukenk and I have not been able to track down where GO:0040023 may be coming from. Do you have any clues?

**Comment by Paul-Davis** [comment]
@chris-grove for WS262 data I find:

GO_term data which is probably fine 
```
GO_term : "GO:0007097"
Alt_id   -O "2017-09-21_11:07:59_kimberly" "GO:0040023" -O "2017-09-21_11:07:59_kimberly"

GO_term : "GO:0040023" -O "2017-09-25_18:15:21_wormpub"

```

and  Phenotype data which probably need updating to use GO:0007097

```
Phenotype : "WBPhenotype:0000511" -O "2017-09-25_17:54:47_wormpub"
Description      -O "2017-09-21_11:07:04_gary" "Variations in the directed movement of the nucleus to a specific location within a cell compar
ed to control." -O "2017-09-21_11:07:04_gary" Person_evidence -O "2017-09-21_11:07:04_gary" "WBPerson2021" -O "2017-09-21_11:07:04_gary"
Description      -O "2017-09-21_11:07:04_gary" "Variations in the directed movement of the nucleus to a specific location within a cell compared to control." -O "2017-09-21_11:07:04_gary" GO_term_evidence -O "2017-09-21_11:07:04_gary" "GO:0040023" -O "2017-09-21_11:07:04_gary"
```

Hope this helps

**Comment by chris-grove** [comment]
@Paul-Davis 
OK, thanks, that helps.

**Comment by vanaukenk** [comment]
Thanks, @Paul-Davis 
I've just double-checked all my files for the WS263 upload and they are okay wrt these terms.
@chris-grove - can we go ahead and close this ticket?  
Thx.

**Comment by chris-grove** [comment]
@vanaukenk 
I'm still waiting to hear back from @Gschindelman about replacing the reference to GO:0040023 in the nuclear positioning variant phenotype term GO_term_evidence

**Comment by Gschindelman** [comment]
Yes I will replace this term
On Nov 16, 2017, at 8:24 AM, Chris Grove <notifications@github.com<mailto:notifications@github.com>> wrote:

GO:0040023

**Comment by Gschindelman** [comment]
sorry…sent too soon..
No worries.  The obsolete terms usually come up when I run an error check before upload.  I will look into why this wasn’t  found.
On Nov 16, 2017, at 9:03 AM, Gary Schindelman <garys@caltech.edu<mailto:garys@caltech.edu>> wrote:

Yes I will replace this term
On Nov 16, 2017, at 8:24 AM, Chris Grove <notifications@github.com<mailto:notifications@github.com>> wrote:

GO:0040023

**Comment by vanaukenk** [comment]
@Gschindelman 
Juancarlos and I looked into the pipeline that updates the go term tables and postgres and found that, due to a change in the header of the GO file, the tables were not being updated for a few months.  GO:0040023 was obsoleted in July of 2017, so that's probably why your checking script didn't catch it.
I believe we've fixed the GO parsing pipeline, so all should be fine there now.
Thx.

**Comment by Gschindelman** [comment]
Yes…Just ran the script and it picked up the obsolete GO terms…will fix for upload…all is good.
Gary
On Nov 16, 2017, at 9:21 AM, vanaukenk <notifications@github.com<mailto:notifications@github.com>> wrote:


@Gschindelman<https://github.com/gschindelman>
Juancarlos and I looked into the pipeline that updates the go term tables and postgres and found that, due to a change in the header of the GO file, the tables were not being updated for a few months. GO:0040023 was obsoleted in July of 2017, so that's probably why your checking script didn't catch it.
I believe we've fixed the GO parsing pipeline, so all should be fine there now.
Thx.
